src/app.rs

```diff
@@ -1,110 +1,148 @@
-                .takes_value(true)
-                .help("Set maximum search depth (default: none)"),
-        )
-        .arg(
-            Arg::with_name("file-type")
+            arg("file-type")
                 .long("type")
                 .short("t")
                 .takes_value(true)
                 .value_name("filetype")
                 .possible_values(&["f", "file", "d", "directory", "s", "symlink"])
-                .hide_possible_values(true)
-                .help("Filter by type: f(ile), d(irectory), s(ymlink)"),
+                .hide_possible_values(true),
         )
         .arg(
-            Arg::with_name("extension")
+            arg("extension")
                 .long("extension")
                 .short("e")
                 .takes_value(true)
-                .value_name("ext")
-                .help("Filter by file extension"),
+                .value_name("ext"),
         )
         .arg(
-            Arg::with_name("color")
+            arg("color")
                 .long("color")
                 .short("c")
                 .takes_value(true)
                 .value_name("when")
                 .possible_values(&["never", "auto", "always"])
-                .hide_possible_values(true)
-                .help(
-                    "When to use color in the output:\n\
-                     never, auto, always (default: auto)",
-                ),
+                .hide_possible_values(true),
         )
         .arg(
-            Arg::with_name("threads")
+            arg("threads")
                 .long("threads")
                 .short("j")
                 .takes_value(true)
-                .value_name("num")
-                .help(
-                    "Set number of threads to use for searching\n\
-                     (default: number of available CPU cores)",
-                ),
+                .value_name("num"),
         )
         .arg(
-            Arg::with_name("max-buffer-time")
+            arg("max-buffer-time")
                 .long("max-buffer-time")
                 .takes_value(true)
-                .hidden(true)
-                .help("the time (in ms) to buffer, before streaming to the console"),
+                .hidden(true),
         )
-        .arg(Arg::with_name("pattern").help("the search pattern, a regular expression (optional)"))
-        .arg(Arg::with_name("path").help("the root directory for the filesystem search (optional)"))
+        .arg(arg("pattern"))
+        .arg(arg("path"))
+}
+
+#[cfg_attr(rustfmt, rustfmt_skip)]
+fn usage() -> HashMap<&'static str, Help> {
+    let mut h = HashMap::new();
+    doc!(h, "hidden"
+        , "Search hidden files and directories"
+        , "Includes, in a search query, any hidden files and directories for the directory of \
+            the search.");
+    doc!(h, "no-ignore"
+        , "Do not respect .(git)ignore files"
+        , "Searches for a pattern match with files and directories, including all defined \
```

Comment:
Same as above: "Show search results from files and directories that would otherwise be ignored by '.*ignore' files."